### DISCUSSION TITLE
Was Remove OSX Files deleted from Tools?

### DISCUSSION THREAD

**Discussion by Phanturian** [OnionUI/Onion#1080] [Q&A]
I just updated to Onion-v4.2.0-RC-snapshot-a5064207, and I noticed that Remove OSX Files is missing from Tools. I'm a Mac user, and this was a handy feature. I was hoping for some clarity as to whether it was removed intentionally, or if it will return in a future update. The Tools menu still says "Favorites, clean files" but the only three options within are generate tools.

![IMG_3378](https://github.com/OnionUI/Onion/assets/135469050/dead1a9c-175d-43b2-bb50-82dcd62688a8)

![IMG_3377](https://github.com/OnionUI/Onion/assets/135469050/37a4bddf-7a27-4e55-9642-7dd437f134a8)

**Comment by ghost** [comment]
You probably know this, but until it comes back you can always use the Terminal with 
_dot_clean /path/to/drive_
command before ejecting the SD card to clean the macOS junk :)

**Reply by Phanturian** [reply]
Thanks, I use dot_clean before I eject my SD every time. I was mostly asking on behalf of some friends I swap builds with who don't have access to Macs. Dot clean usually works fine but sometimes they sneak through. More problematic is the creation of DS_Store and __MACOSX when zipping files on Mac. These show up on the Miyoo Mini (Plus) as a hidden folder at the top of your game list. I have to zip things using Google Drive or Telegram to avoid generating these files.

You mention "until it comes back," so I was wondering if you know it will return for certain. No worries either way. Thanks to everyone working on Onion OS. I can't believe how far it's come along.

**Reply by a8bit** [reply]
As an aside, Keka has the option to ignore mac resource files when compressing.

**Reply by Phanturian** [reply]
Thank you so much! I just downloaded it and it looks like this is what I needed.

**Comment by forsaoss** [comment]
@Phanturian -- to deal with the zip nonsense, I'm not sure how you're zipping files but if you're using the terminal and `zip` you can pass it the `--no-extra` flag (or `-X` short flag) to skip adding extended attributes to the zip whcih should skip the creation of the __MACOSX nonsense.  See the manpage for zip about this option.

Another way that I personally prefer is to clean up the source files on the Mac before compression.  Use a combination of `xattr` for stripping the extended attributes and `find` for finding and deleting nonsense like the .DS_Store files. This assumes you are working in the terminal (or whatever similar program) and avoiding tools like Finder that gratuitously add macOS-specific nonsense to files and folders.

For example, before zipping a source tree named `source` for this example, I'd do something like the following:
```
xattr -c -r source
find source -type f -name .DS_Store -delete
# proceed with compression etc etc
```

See the manpages for both of those utilities if you're unfamiliar with them.

**Reply by Phanturian** [reply]
Thanks for the info. I'm rather inexperienced with Terminal but I'm learning. I only zip files with Terminal when I want to break up large files into multiple zip files. I'll look into your advice. Thanks again.

**Comment by michaljuris** [comment]
Actually, the tool was removed in this commit:

https://github.com/OnionUI/Onion/commit/9e8b32fcd20dd45e5aef66b849ec7f089e21490e by @Aemiii91 

Since the description of the commit is _**FIX: Remove unnecessary dot-clean tool**_ I guess it is not necessary anymore. I haven't tested it yet though.

**Reply by Phanturian** [reply]
Most people probably don't use it, and I wonder if the feature was the victim of decluttering. Would be nice to have it back.

**Comment by schmurtzm** [comment]
Hi, we have initially created the dot cleaner because these mac files were listed in MainUI (in other words it was listed in the roms lists). 

It's not the case anymore, MainUI doesn't show them anymore and when you put your SD card in your mac, these files are created again and they are not visible in Finder.

Of course it's still possible to add the tool but it will stay a lot less efficient than running a cleaning tool from your mac (The cleaning script takes a lot of time to be executed and for many users seems freezed so as it is useless it seems to be a good idea to remove it to have a cleaner menu a keep place for potential future items).

I have no position on this (really 😅), we could add it again there is technically no problem, but is there someone who can explain me what's the point of having this tool in Onion now that it's no longer a problem for the roms list?

**Reply by Phanturian** [reply]
Thanks for the response. I didn't realize that dot files were now hidden in MainUI. I have been part of a small group that builds and shares, and we have both Mac and PC users. Anytime a Mac user shares with PC users, it seems dot files show up. The PC users would normally go through and delete them all by hand, but the introduction of the Tool made it very convenient for them.

As a Mac user, it isn't important to me specifically, but I couldn't understand why this feature was removed. While I now understand the reasoning, I also don't feel it was causing problems. If the final decision is to leave this feature out, perhaps the words "Clean Files" should be removed from the Tools description.

Thanks to the whole Onion team for their hard work and effort. Tinkering with the Miyoo handhelds has been such a fun endeavor over the past year.
